docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/AnomalyDetection/RandomizedPcaSample.cs

```diff
@@ -0,0 +1,85 @@
+#pragma warning disable 649
+        private class Result
+        {
+            // Outlier gets false while inlier has true.
+            public bool PredictedLabel;
+            // Outlier gets smaller score.
+            public float Score;
+        }
+#pragma warning restore 649
+
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            var samples = new List<DataPoint>()
+            {
+                new DataPoint(){ Features= new float[3] {1, 0, 0} },
+                new DataPoint(){ Features= new float[3] {0, 2, 1} },
+                new DataPoint(){ Features= new float[3] {1, 2, 3} },
+                new DataPoint(){ Features= new float[3] {0, 1, 0} },
+                new DataPoint(){ Features= new float[3] {0, 2, 1} },
+                new DataPoint(){ Features= new float[3] {-100, 50, -100} }
+            };
+
+            // Convert native C# class to IDataView, a consumble format to ML.NET functions.
+            var data = mlContext.Data.LoadFromEnumerable(samples);
+
+            // Create an anomaly detector. Its underlying algorithm is randomized PCA.
+            var pipeline = mlContext.AnomalyDetection.Trainers.RandomizedPca(featureColumnName: nameof(DataPoint.Features), rank: 1, center: false);
+
+            // Train the anomaly detector.
+            var model = pipeline.Fit(data);
+
+            // Apply the trained model on the training data.
+            var transformed = model.Transform(data);
+
+            // Read ML.NET predictions into C# class.
+            var results = mlContext.Data.CreateEnumerable<Result>(transformed, reuseRowObject: false).ToList();
+
+            // Let's go through all predictions.
+            for (int i = 0; i < samples.Count; ++i)
+            {
+                // The i-th example's prediction result.
+                var result = results[i];
+
+                // The i-th example's feature vector in text format.
+                var featuresInText = string.Join(',', samples[i].Features);
+
+                if (result.PredictedLabel)
+                    // The i-th sample is predicted as an inlier.
+                    Console.WriteLine("The {0}-th example with features [{1}] is an inlier with a score of being inlier {2}",
+                        i, featuresInText, result.Score);
+                else
+                    // The i-th sample is predicted as an outlier.
+                    Console.WriteLine("The {0}-th example with features [{1}] is an outlier with a score of being inlier {2}",
+                        i, featuresInText, result.Score);
+            }
```

Comment:
please put in comment below what would the output of this look.  (The results of those WriteLine)

Reply:
Sounds good. Please see iteration 3.

---
In reply to: [261047636](https://github.com/dotnet/machinelearning/pull/2780#discussion_r261047636) [](ancestors = 261047636)